go/documentation_generation.go

```diff
@@ -56,52 +58,46 @@ func detectErrorCodeChanges(ctx context.Context, prInfo prInformation, client *g
 	return false, nil
 }
 
-func cloneVitessAndGenerateErrors(prInfo prInformation) (string, error) {
-	_, err := execCmd("", "git", "clone", fmt.Sprintf("git@github.com:%s/%s.git", prInfo.repoOwner, prInfo.repoName), "/tmp/vitess")
-	if err != nil && !strings.Contains(err.Error(), "already exists and is not an empty directory") {
+func cloneVitessAndGenerateErrors(ctx context.Context, vitess *git.Repo, prInfo prInformation) (string, error) {
+	if err := vitess.Clone(ctx); err != nil {
 		return "", errors.Wrapf(err, "Failed to clone repository %s/%s to generate error code on Pull Request %d", prInfo.repoOwner, prInfo.repoName, prInfo.num)
 	}
 
 	// Clean the repository
-	_, err = execCmd("/tmp/vitess", "git", "clean", "-fd")
-	if err != nil {
+	if err := vitess.Clean(ctx); err != nil {
 		return "", errors.Wrapf(err, "Failed to clean the repository %s/%s to generate documentation %d", prInfo.repoOwner, prInfo.repoName, prInfo.num)
 	}
 
-	_, err = execCmd("/tmp/vitess", "git", "fetch", "origin", fmt.Sprintf("refs/pull/%d/head", prInfo.num))
-	if err != nil {
+	if err := vitess.FetchRef(ctx, "origin", fmt.Sprintf("refs/pull/%d/head", prInfo.num)); err != nil {
 		return "", errors.Wrapf(err, "Failed to fetch Pull Request %s/%s#%d to generate error code", prInfo.repoOwner, prInfo.repoName, prInfo.num)
 	}
 
-	_, err = execCmd("/tmp/vitess", "git", "checkout", "FETCH_HEAD")
-	if err != nil {
+	if err := vitess.Checkout(ctx, "FETCH_HEAD"); err != nil {
 		return "", errors.Wrapf(err, "Failed to checkout on Pull Request %s/%s#%d to generate error code", prInfo.repoOwner, prInfo.repoName, prInfo.num)
 	}
 
-	vterrorsgenVitessBytes, err := execCmd("/tmp/vitess", "go", "run", "./go/vt/vterrors/vterrorsgen")
+	vterrorsgenVitessBytes, err := shell.NewContext(ctx, "go", "run", "./go/vt/vterrors/vterrorsgen").InDir("/tmp/vitess").Output()
 	if err != nil {
 		return "", errors.Wrapf(err, "Failed to run ./go/vt/vterrors/vterrorsgen on Pull Request %s/%s#%d to generate error code", prInfo.repoOwner, prInfo.repoName, prInfo.num)
 	}
 	return string(vterrorsgenVitessBytes), err
 }
 
-func cloneWebsiteAndGetCurrentVersionOfDocs(prInfo prInformation) (string, error) {
-	_, err := execCmd("/tmp", "git", "clone", fmt.Sprintf("https://github.com/%s/website", prInfo.repoOwner))
-	if err != nil && !strings.Contains(err.Error(), "already exists and is not an empty directory") {
+func cloneWebsiteAndGetCurrentVersionOfDocs(ctx context.Context, website *git.Repo, prInfo prInformation) (string, error) {
+	if err := website.Clone(ctx); err != nil {
 		return "", errors.Wrapf(err, "Failed to clone repository vitessio/website to generate error code on Pull Request %d", prInfo.num)
 	}
 
-	_, err = execCmd("/tmp/website", "git", "pull")
-	if err != nil {
+	if err := website.Clean(ctx); err != nil {
 		return "", errors.Wrapf(err, "Failed to fetch vitessio/website to generate error code on Pull Request %d", prInfo.num)
 	}
```

Comment:
I think we might want to keep doing a `git pull` here to update our view of the `website` repository, a `clean` + a `git pull` might be even better.

Reply:
yep this got dropped by mistake!